alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -58,56 +54,52 @@
+    public DefaultProviderDataAccessor(ProviderProjectRepository providerProjectRepository, ProviderUserProjectRelationRepository providerUserProjectRelationRepository, ProviderUserRepository providerUserRepository,
+        ConfigurationAccessor configurationAccessor) {
         this.providerProjectRepository = providerProjectRepository;
         this.providerUserProjectRelationRepository = providerUserProjectRelationRepository;
         this.providerUserRepository = providerUserRepository;
+        this.configurationAccessor = configurationAccessor;
     }
 
     @Override
     @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public Optional<ProviderProject> findFirstByHref(String href) {
-        return providerProjectRepository.findFirstByHref(href).map(this::convertToProjectModel);
-    }
-
-    @Override
-    @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public Optional<ProviderProject> findFirstByName(String name) {
-        return providerProjectRepository.findFirstByName(name).map(this::convertToProjectModel);
-    }
-
-    @Override
-    @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public List<ProviderProject> findByProviderName(String providerName) {
-        return providerProjectRepository.findByProvider(providerName)
-                   .stream()
-                   .map(this::convertToProjectModel)
-                   .collect(Collectors.toList());
+    public List<ProviderProject> findByProviderConfigName(String providerConfigName) {
+        try {
+            Optional<Long> optionalConfigId = configurationAccessor.getProviderConfigurationByName(providerConfigName)
+                                                  .map(ConfigurationModel::getConfigurationId);
+            if (optionalConfigId.isPresent()) {
+                return providerProjectRepository.findByProviderConfigId(optionalConfigId.get())
+                           .stream()
+                           .map(this::convertToProjectModel)
+                           .collect(Collectors.toList());
+            }
+        } catch (AlertDatabaseConstraintException ignored) {
+        }
+        return List.of();
     }
 
     @Override
     @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public List<ProviderProject> findByProviderKey(ProviderKey providerKey) {
-        return providerProjectRepository.findByProvider(providerKey.getUniversalKey())
+    public List<ProviderProject> findByProviderConfigId(Long providerConfigId) {
+        return providerProjectRepository.findByProviderConfigId(providerConfigId)
                    .stream()
                    .map(this::convertToProjectModel)
                    .collect(Collectors.toList());
     }
 
     @Override
-    public void deleteProjects(ProviderKey providerKey, Collection<ProviderProject> providerProjects) {
+    public void deleteProjects(Collection<ProviderProject> providerProjects) {
```

Comment:
Are we missing `@Transactional` on this method?

Reply:
The class is annotated with `@Transactional`, so this method will be. The read committed methods just override the class level annotation.